sys-devel/libtool/files/libtool-2.4.7-darwin.patch

```diff
@@ -0,0 +1,21 @@
+Remove bad command on darwin
+
+It causes linker error "undefined symbol: main", but the command defined just above it, works fine.
+
+https://bugs.gentoo.org/758167
+
+--- a/m4/libtool.m4
++++ b/m4/libtool.m4
+@@ -1125,12 +1125,6 @@ m4_defun([_LT_DARWIN_LINKER_FEATURES],
+     _LT_TAGVAR(module_cmds, $1)="\$CC \$allow_undefined_flag -o \$lib -bundle \$libobjs \$deplibs \$compiler_flags$_lt_dsymutil"
+     _LT_TAGVAR(archive_expsym_cmds, $1)="$SED 's|^|_|' < \$export_symbols > \$output_objdir/\$libname-symbols.expsym~\$CC -dynamiclib \$allow_undefined_flag -o \$lib \$libobjs \$deplibs \$compiler_flags -install_name \$rpath/\$soname \$verstring $_lt_dar_single_mod$_lt_dar_export_syms$_lt_dsymutil"
+     _LT_TAGVAR(module_expsym_cmds, $1)="$SED -e 's|^|_|' < \$export_symbols > \$output_objdir/\$libname-symbols.expsym~\$CC \$allow_undefined_flag -o \$lib -bundle \$libobjs \$deplibs \$compiler_flags$_lt_dar_export_syms$_lt_dsymutil"
+-    m4_if([$1], [CXX],
+-[   if test yes != "$lt_cv_apple_cc_single_mod"; then
```

Comment:
I don't suppose e.g. `export lt_cv_apple_cc_single_mod=no` in the ebuild does the same job?

Would really like to avoid a (doubly-)conditional patch.

Reply:
I should try the patch with gcc, perhaps it doesn't break it, then I can remove one condition. Same for the other condition, I didn't check whether it doesn't break normal linux.

I'll try lt_cv_apple_cc_single_mod first though

Reply:
lt_cv_apple_cc_single_mod set in profile works, thanks